source/fundamentals/promises.txt

```diff
@@ -0,0 +1,147 @@
+======================
+Promises and Callbacks
+======================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecols
+
+Overview
+--------
+
+NodeJS driver uses the asynchronous Javascript API to communicate with
+your MongoDB cluster.
+
+Asynchronous Javascript allows you to execute operations without waiting for 
+the processing thread to become free. This helps prevent your application
+from becoming unresponsive to web UI rendering or network requests when
+executing long-running operations. For more information about asynchronous 
```

Comment:
This is just personal preference, but we could remove the "For more information about asynchronous Javascript, see the MDN ..." line and just link to the MDN link in the first sentence of the paragraph: 
"`Asynchronous Javascript<developer.mozzila.org/../Asynchronous/>`_ allows you to execute operations without waiting for 
the processing thread to become free"

Reply:
Thanks, I appreciate your thoughts on streamlining this. I intended to notify the user where the link goes, especially since it potentially draws users away from the page and our domain. The structure I'm using seems aligned with that goal:
1) Describe the term
2) Explain when it's used
3) Direct to other resources for further learning